src/main/scala/io/iohk/ethereum/jsonrpc/QAService.scala

```diff
@@ -44,11 +44,15 @@ class QAService(
   def generateCheckpoint(
       req: GenerateCheckpointRequest
   ): ServiceResponse[GenerateCheckpointResponse] = {
-    Task {
-      val hash = req.blockHash.getOrElse(blockchain.getBestBlock().hash)
-      val checkpoint = generateCheckpoint(hash, req.privateKeys)
-      syncController ! NewCheckpoint(hash, checkpoint.signatures)
-      Right(GenerateCheckpointResponse(checkpoint))
+    val hash = req.blockHash orElse blockchain.getBestBlock().map(_.hash)
```

Comment:
I know we don't have a style set in stone yet but can we agree that the infix notation is not good scala? :)